prometheus/prometheus.go

```diff
@@ -177,17 +193,20 @@ func (p *PrometheusSink) IncrCounter(parts []string, val float32) {
 
 func (p *PrometheusSink) IncrCounterWithLabels(parts []string, val float32, labels []metrics.Label) {
 	key, hash := p.flattenKey(parts, labels)
-	g, ok := p.counters.Load(hash)
+	pc, ok := p.counters.Load(hash)
 	if !ok {
-		g = prometheus.NewCounter(prometheus.CounterOpts{
+		c := prometheus.NewCounter(prometheus.CounterOpts{
 			Name:        key,
 			Help:        key,
 			ConstLabels: prometheusLabels(labels),
 		})
-		p.counters.Store(hash, g)
+		pc = &PrometheusCounter{
+			c, time.Now(),
+		}
+		p.counters.Store(hash, pc)
 	}
-	g.(prometheus.Counter).Add(float64(val))
-	p.updates.Store(hash, time.Now())
+	pc.(*PrometheusCounter).Add(float64(val))
+	pc.(*PrometheusCounter).updatedAt = time.Now()
```

Comment:
Is this a race condition?  It feels like it might be, in that we're now modifying the contents of memory without any synchronization.  I think we should be populating a private PrometheusCounter value and then calling Store on it, rather than calling Store and then modifying the value stored after the fact.

Reply:
Ah, right, good call. I'll update this.